bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/DanfossAirUnitHandler.java

```diff
@@ -0,0 +1,165 @@
+                if (channel.getWriteAccessor() != null) {
+                    updateState(channelUID, channel.getWriteAccessor().access(danfossAirUnit, command));
+                }
+            } catch (IllegalArgumentException e) {
+                logger.info("{}", e.getMessage());
+            } catch (IOException ioe) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, ioe.getMessage());
+            } catch (Exception e) {
+                logger.info("Error updating channel {}: {}", channelUID, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        config = getConfigAs(DanfossAirUnitConfiguration.class);
+        valueCache = new ValueCache(config.updateUnchangedValuesEveryMillis);
+        try {
+            hrv = new DanfossAirUnit(InetAddress.getByName(config.host), 30046);
+        } catch (UnknownHostException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Unknown host: " + config.host);
+            return;
+        }
+
+        scheduler.execute(() -> {
+            try {
+                thing.setProperty(PROPERTY_UNIT_NAME, hrv.getUnitName());
+                thing.setProperty(PROPERTY_SERIAL, hrv.getUnitSerialNumber());
+                pollingJob = scheduler.scheduleWithFixedDelay(this::updateAllChannels, 5, config.refreshInterval,
+                        TimeUnit.SECONDS);
+                updateStatus(ThingStatus.ONLINE);
+
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+            }
+        });
+    }
+
+    private void updateAllChannels() {
+        if (hrv == null) {
+            return;
+        }
+        DanfossAirUnit danfossAirUnit = hrv;
+        logger.debug("Updating DanfossHRV data '{}'", getThing().getUID());
+
+        try {
+            for (Channel channel : Channel.values()) {
+                if (Thread.interrupted()) {
+                    logger.debug("Polling thread interrupted...");
+                    return;
+                }
+                try {
+                    updateState(channel.getGroup().getGroupName(), channel.getChannelName(),
+                            channel.getReadAccessor().access(danfossAirUnit));
+                } catch(UnexpectedResponseValueException e) {
+                    updateState(channel.getGroup().getGroupName(), channel.getChannelName(), UnDefType.UNDEF);
+                    logger.debug("Cannot update channel {}: an unexpected or invalid response has been received from the air unit: {}",
+                            channel.getChannelName(), e.getMessage());
+                } catch (IOException e) {
+                    updateState(channel.getGroup().getGroupName(), channel.getChannelName(), UnDefType.UNDEF);
```

Comment:
At first glance, this catches exception before the outer `catch` can handle them. Should this `catch` update any status?